src/mono/mono/metadata/object.c

```diff
@@ -5143,11 +5145,12 @@ mono_unhandled_exception_checked (MonoObjectHandle exc, MonoError *error)
 #else
 		field = mono_class_get_field_from_name_full (mono_defaults.appcontext_class, "UnhandledException", NULL);
```

Comment:
Could do the same `if (!inited)` thing here, too, but this is on the unhandled exception failure path, so it won't run more than once anyway
```suggestion
		static gboolean inited;
		if (!inited) {
			field = mono_class_get_field_from_name_full (mono_defaults.appcontext_class, "UnhandledException", NULL);
			inited = TRUE;
		}
```